src/PiWeb.Import.Sdk/Modules/ImportFormat/FileSourceExtensions.cs

```diff
@@ -0,0 +1,74 @@
+﻿#region copyright
+
+/* * * * * * * * * * * * * * * * * * * * * * * * * */
+/* Carl Zeiss Industrielle Messtechnik GmbH        */
+/* Softwaresystem PiWeb                            */
+/* (c) Carl Zeiss 2024                             */
+/* * * * * * * * * * * * * * * * * * * * * * * * * */
+
+#endregion
+
+namespace Zeiss.PiWeb.Import.Sdk.Modules.ImportFormat;
+
+#region usings
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+#endregion
+
+/// <summary>
+/// Provides methods to work with <see cref="IFileSource"/> more easily.
+/// </summary>
+public static class FileSourceExtensions
+{
+    #region methods
+
+    /// <summary>
+    /// Checks if <paramref name="fileSource"/> has the requested file extension <paramref name="extension"/>.
+    /// </summary>
+    /// <param name="fileSource">The <see cref="IFileSource"/> to check.</param>
+    /// <param name="extension">The file extension (including the period ".").</param>
+    /// <returns><c>true</c> if the file has the extension; otherwise <c>false</c>.</returns>
+    public static bool HasExtension(this IFileSource fileSource, string extension)
+    {
+        return string.Equals(fileSource.Extension, extension, StringComparison.OrdinalIgnoreCase);
```

Comment:
I think this method (and all similar methods) should also work correctly when passing extensions containing multiple .-characters. We need to check whether the name ends with the given extension. If the given extension does not begin with a .-character we also need to check whether the name contains a .-character at the correct position. I.e. the filename "foo.bar" does not have the extension "ar". It does have the extension "bar" or ".bar" though.